### ISSUE TITLE
TypeError when testcafe is stopping

### ISSUE THREAD

**Issue by TauPan** [DevExpress/testcafe-live#21]
```javascript
Stopping TestCafe Live...
/home/taupan/nobackup/git/myproject/node_modules/testcafe-live/lib/test-run-controller.js:175
                    testWrapper.testRunWrappers.forEach(w => w.finish());
                                                               ^

TypeError: w.finish is not a function
    at testWrapper.testRunWrappers.forEach.w (/home/taupan/nobackup/git/myproject/node_modules/testcafe-live/lib/test-run-controller.js:175:64)
    at Array.forEach (<anonymous>)
    at Timeout.setTimeout (/home/taupan/nobackup/git/myproject/node_modules/testcafe-live/lib/test-run-controller.js:175:49)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

(This is an actual backtrace, but I redacted the directory name to be "myproject" .)

I see these kinds of errors sometimes when testcafe-live sees a test failure, in which case they occur persistently with every run.

Command line is
```
cross-env DJANGO_SETTINGS_MODULE=myproject.settings_testcafe testcafe-live chromium myproject/js/testcafe/"test_*.js" --hostname localhost --screenshots myproject/js/testcafe/screenshots --quarantine-mode --screenshots-on-fails --app-init-delay 3000 --app "yarn testcafe-db; yarn testcafe-server --noreload || sleep 2; netcat -z -w 2 localhost 8003" --concurrency 8
```
(Edit: Inserted command and browser.)

**Comment by AndreyBelym** [comment]
Hi @TauPan, I need some time to reproduce the problem. I guess that concurrency or quarantine mode can cause the failure, could you please try to run tests without enabling these features?

**Comment by TauPan** [comment]
Hi, I'm trying to run it without concurrency or quarantine... I just noticed I gave the wrong browser above. I'm usually running with `chromium:headless`. Might that be relevant as well?

**Comment by shinriyo** [comment]
mine also stopped TCP? error.

```
TestCafe Live watches the files and reruns
the tests once you've saved your changes.

You can use the following keys in the terminal:
'ctrl+s' - stop current test run;
'ctrl+r' - restart current test run;
'ctrl+w' - turn off/on watching;
'ctrl+c' - close browsers and terminate the process.


Test run is starting...
 Running tests in:
 - Chrome 66.0.3359 / Mac OS X 10.13.2

 Getting Started

Stopping TestCafe Live...
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TCP.onread (net.js:659:25)
Emitted 'error' event at:
    at Socket.onerror (_stream_readable.js:696:12)
    at Socket.emit (events.js:182:13)
    at Socket.EventEmitter.emit (domain.js:442:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:174:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Klaster1** [comment]
I get the same error when running testcafe and testcafe-live with Node 10, but if I use Node 9 binary, everything works fine or at least stable enough.

**Comment by ColCh** [comment]
@Klaster1 @shinriyo may be tracked in https://github.com/DevExpress/testcafe/issues/2711

**Comment by AndreyBelym** [comment]
Hi @shinriyo @Klaster1 @myxious @amilajack @Itran273 @vickvasquez @oshoram @MoritzR the issue with ECONNRESET must be fixed in `testcafe@0.22.0`, could you please try it on your machines?

**Comment by Klaster1** [comment]
@AndreyBelym I couldn't reproduce the issue anymore, looks like the fix works. Great work!

**Comment by MoritzR** [comment]
@AndreyBelym Did a few runs on node 10.1.0 with both testcafe v0.21.1 and v0.22.0 and could confirm that it seems to be fixed with v0.22.0, thanks!

**Comment by AndreyBelym** [comment]
I can't reproduce the issue with Live mode integrated into `testcafe@1.0.0`: [TestCafe 1.0.0 Released: Live mode](https://devexpress.github.io/testcafe/blog/testcafe-v1-0-0-released.html#live-mode-3215). If you still can reproduce with `testcafe@1.0.0`, please create a [new bug report](https://github.com/DevExpress/testcafe/issues/new?template=bug-report.md) in the main TestCafe repository.
